crate/server/src/core/operations/validate.rs

```diff
@@ -0,0 +1,586 @@
+        println!("CHAIN STRUCTURE IS VALID IN: validate_chain_structure");
+        KResult::Ok((ValidityIndicator::Valid, count))
+    } else {
+        println!("CHAIN STRUCTURE IS INVALID IN: validate_chain_structure");
+        KResult::Ok((ValidityIndicator::Invalid, count))
+    }
+}
+
+// If fetching a certificate fails, the method reports the first error happening.
+// Otherwise it returns a vector of certificates.
+async fn certificates_by_uid(
+    unique_identifiers: Vec<UniqueIdentifier>,
+    kms: &KMS,
+    user: &str,
+    params: Option<&ExtraDatabaseParams>,
+) -> KResult<Vec<Vec<u8>>> {
+    let res = join_all(unique_identifiers.iter().map(|unique_identifier| async {
+        let unique_identifier = unique_identifier
+            .as_str()
+            .expect("as_str returned None in certificates_by_uid");
+        certificate_by_uid(unique_identifier, kms, user, params).await
+    }))
+    .await;
+
+    // filtering errors
+    let is_err = res.iter().all(|x| x.is_err());
+
+    // checking if there are any errors
+    if !is_err {
+        KResult::Ok(
+            res.iter()
+                .map(|x| x.clone().expect("safe unwrap"))
+                .collect(),
+        )
+    } else {
+        KResult::Err(KmsError::from(KmipError::ObjectNotFound(
+            "There is a Kms Certificate UID that cannot be retrieved".to_string(),
+        )))
+    }
+}
+
+// Fetches a certificate. If it fails, returns the according error
+async fn certificate_by_uid(
+    unique_identifier: &str,
+    kms: &KMS,
+    user: &str,
+    params: Option<&ExtraDatabaseParams>,
+) -> KResult<Vec<u8>> {
+    let uid_own = retrieve_object_for_operation(
+        unique_identifier,
+        ObjectOperationType::Validate,
+        kms,
+        user,
+        params,
+    )
+    .await;
+    match uid_own {
+        Err(e) => KResult::Err(e),
+        Ok(kms_object) => {
+            if let Object::Certificate {
+                certificate_type: _,
+                certificate_value,
+            } = kms_object.object
+            {
+                KResult::Ok(certificate_value)
+            } else {
+                KResult::Err(KmsError::from(KmipError::InvalidKmipObject(
+                    cosmian_kmip::kmip::kmip_operations::ErrorReason::Invalid_Object_Type,
+                    String::from("Requested a Certificate Object, got a ")
+                        + &kms_object.object.object_type().to_string(),
+                )))
+            }
+        }
+    }
+}
+
+fn validate_chain_date(
+    certificates: &mut [Vec<u8>],
+    date: Option<String>,
+) -> KResult<ValidityIndicator> {
+    println!("Looking for a date to compare with");
+    let current_date = {
+        if let Some(date) = date {
+            println!("Converting date from input");
+            Asn1Time::from_str(date.as_str())
+        } else {
+            println!("Getting current date");
+            Asn1Time::days_from_now(0)
+        }
+    }?;
```

Comment:
No need for an additional block since `if` *returns*:
```suggestion
    let current_date = if let Some(date) = date {
            println!("Converting date from input");
            Asn1Time::from_str(date.as_str())
        } else {
            println!("Getting current date");
            Asn1Time::days_from_now(0)
        }?;
```